### ISSUE TITLE
Angular 10 Update Generic type 'ModuleWithProviders' requires 1 type argument(s).

### ISSUE THREAD

**Issue by PauloPeres** [oferh/ng2-completer#456]
WHERE: 
@akveo/ng2-completer/src/modules/ng2-completer.module.d.ts:3:23 - error TS2314: Generic type 'ModuleWithProviders<T>' requires 1 type argument(s).
3     static forRoot(): ModuleWithProviders;

According to Angular 10 it's now necessary to have a generic type 
https://angular.io/guide/migration-module-with-providers

**Comment by vamidi** [comment]
Hopefully a fix for this soon!

**Comment by PauloPeres** [comment]
@vamidi sent a pull request but looks like no one answered yet, not sure if this rep has people maintaining.

**Comment by vamidi** [comment]
So unfortunate that there is no update there are a few frameworks that rely on this.

**Comment by aaron-harvey** [comment]
I used [patch-package](https://www.npmjs.com/package/patch-package) with
```js
diff --git a/node_modules/ng2-completer/src/modules/ng2-completer.module.d.ts b/node_modules/ng2-completer/src/modules/ng2-completer.module.d.ts
index f1dd11a..860a84c 100644
--- a/node_modules/ng2-completer/src/modules/ng2-completer.module.d.ts
+++ b/node_modules/ng2-completer/src/modules/ng2-completer.module.d.ts
@@ -1,5 +1,5 @@
 import { ModuleWithProviders } from "@angular/core";
 export declare class Ng2CompleterModule {
-    static forRoot(): ModuleWithProviders;
-    static forChild(): ModuleWithProviders;
+    static forRoot(): ModuleWithProviders<Ng2CompleterModule>;
+    static forChild(): ModuleWithProviders<Ng2CompleterModule>;
 }
```
As a stop gap

Hopefully a more permanent fix is merged soon.

**Comment by vamidi** [comment]
Yes hopefully. I have created my own repo for now until the fix comes up. https://github.com/vamidi/ng2-completer

**Comment by BruneXX** [comment]
Thanks @aaron-harvey that tool is awesome! :)

**Comment by dilankamr** [comment]
Hi All,

I have forked the project and **fixed the issue**, If anyone wants the fix, replace your `"ng2-completer": "^9.0.1"` dependency line in **package.json** with this  line: `"ng2-completer": "git+https://github.com/dilankamr/ng2-completer.git"`

Or run the command: `npm install git+https://github.com/dilankamr/ng2-completer.git --save` 

And do an npm install. 

Then, your project will run nicely without any issue. :)

Thanks
